### PR TITLE
Hotfix for nullpay logging on mobile platforms

### PR DESCRIPTION
**Pull request by mattraffel**
This PR is to get an artifact to the vcx team with logging disabled in libnullpay

### CODE REVIEW

File: libnullpay/src/lib.rs [hunk 1]
```diff
@@ -23,9 +23,10 @@ use std::ffi::CString;
 
 #[no_mangle]
 pub extern fn nullpay_init() -> ErrorCode {
-    if let Err(err) = utils::logger::LibnullpayLogger::init() {
```
Comment (jovfer) [inline libnullpay/src/lib.rs]: I suggest to use platform specific directive instead of commenting code

File: libnullpay/src/utils/logger.rs [hunk 2]
```diff
@@ -1,3 +1,6 @@
+#![allow(dead_code)]
```
Comment (jovfer) [inline libnullpay/src/utils/logger.rs]: Please add reference to ticket in Jira devoted to removing this hotfix.
If possible and only few functions are unused, please move global directive to particular functions.

File: libnullpay/src/lib.rs (hunk 1 above)
Comment (jovfer) [reply libnullpay/src/lib.rs]: Also please add comment and reference to ticket/bug in Jira about resolving mobile platform problem

Comment (mattraffel) [reply libnullpay/src/lib.rs]: I added directives.

File: libnullpay/src/utils/logger.rs (hunk 2 above)
Comment (jovfer) [reply libnullpay/src/utils/logger.rs]: After new changes below it's seems like these 2 allows are not required now, isn't it?

Comment (mattraffel) [reply libnullpay/src/utils/logger.rs]: agreed.

**Pull request merged**